lib/Sema/TypeCheckDecl.cpp

```diff
@@ -482,6 +482,47 @@ static void checkInheritanceClause(
   }
 }
 
+// Check for static properties that produce empty option sets
+static void checkForEmptyOptionSet(const VarDecl *VD, TypeChecker &tc) {
+  if (!VD->isStatic() || !VD->isLet())
+    return;
+  
+  auto DC = VD->getDeclContext();
+  
+  auto *optionSetProto = tc.Context.getProtocol(KnownProtocolKind::OptionSet);
+  auto protocolConformance = tc.containsProtocol(DC->getSelfTypeInContext(), optionSetProto, DC, /*Flags*/None);
+  if (!protocolConformance)
+    return;
+  
+  if (!VD->getType()->isEqual(DC->getSelfTypeInContext()))
```

Comment:
This is a cheaper check than the conformance one, so maybe it should go first. (Not like you should have known this ahead of time.)